### PR TITLE
Add denial exceptions classes in common code

### PR DESCRIPTION
**Pull request by Tetraquark**
Add denial exceptions classes in common code (#4)

### CODE REVIEW

File: permissions/src/commonMain/kotlin/dev/icerock/moko/permissions/DeniedExceptions.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package dev.icerock.moko.permissions
+
+open class DeniedException(val permission: Permission, message: String?) : Exception(message) {
+    constructor(permission: Permission) : this(permission, null)
```
Comment (Alex009) [inline permissions/src/commonMain/kotlin/dev/icerock/moko/permissions/DeniedExceptions.kt]: you can just use default value at `message` arg:
```
open class DeniedException(val permission: Permission, message: String? = null) : Exception(message)
```

File: permissions/src/commonMain/kotlin/dev/icerock/moko/permissions/DeniedExceptions.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+}
+
+class DeniedAlwaysException(permission: Permission, message: String?) : DeniedException(permission, message) {
+    constructor(permission: Permission) : this(permission, null)
```
Comment (Alex009) [inline permissions/src/commonMain/kotlin/dev/icerock/moko/permissions/DeniedExceptions.kt]: also default value can be used

File: permissions/src/iosMain/kotlin/dev/icerock/moko/permissions/PermissionsController.kt [hunk 3]
```diff
@@ -22,8 +22,8 @@ actual class PermissionsController {
             Permission.GALLERY -> provideGalleryPermission()
             Permission.CAMERA -> provideCameraPermission()
             Permission.STORAGE -> { } // not needed any permissions to storage
-            Permission.LOCATION -> provideLocationPermition()
-            Permission.COARSE_LOCATION -> provideLocationPermition()
+            Permission.LOCATION -> provideLocationPermition(permission)
```
Comment (Alex009) [inline permissions/src/iosMain/kotlin/dev/icerock/moko/permissions/PermissionsController.kt]: `provideLocationPermition` -> `provideLocationPermission`

Comment (Tetraquark) [reply permissions/src/iosMain/kotlin/dev/icerock/moko/permissions/PermissionsController.kt]: Fixed.

File: permissions/src/commonMain/kotlin/dev/icerock/moko/permissions/DeniedExceptions.kt (hunk 2 above)
Comment (Tetraquark) [reply permissions/src/commonMain/kotlin/dev/icerock/moko/permissions/DeniedExceptions.kt]: Fixed.

File: permissions/src/commonMain/kotlin/dev/icerock/moko/permissions/DeniedExceptions.kt (hunk 1 above)
Comment (Tetraquark) [reply permissions/src/commonMain/kotlin/dev/icerock/moko/permissions/DeniedExceptions.kt]: Fixed.

**Pull request merged**